va/wayland/va_wayland_drm.c

```diff
@@ -239,46 +226,80 @@ va_wayland_drm_create(VADisplayContextP pDisplayContext)
     wl_drm_ctx->queue = wl_display_create_queue(ctx->native_dpy);
     if (!wl_drm_ctx->queue) {
         va_wayland_error("could not create Wayland event queue");
-        return false;
+        goto end;
     }
 
     wrapped_display = wl_proxy_create_wrapper(ctx->native_dpy);
     if (!wrapped_display) {
         va_wayland_error("could not create Wayland proxy wrapper");
-        return false;
+        goto end;
     }
 
     /* All created objects will inherit this queue */
     wl_proxy_set_queue((struct wl_proxy *) wrapped_display, wl_drm_ctx->queue);
     wl_drm_ctx->registry = wl_display_get_registry(wrapped_display);
-    wl_proxy_wrapper_destroy(wrapped_display);
-    wl_registry_add_listener(wl_drm_ctx->registry, &registry_listener, wl_drm_ctx);
+    if (!wl_drm_ctx->registry) {
+        va_wayland_error("could not create wl_registry");
+        goto end;
+    }
+    if (wl_registry_add_listener(wl_drm_ctx->registry, &registry_listener, wl_drm_ctx) != 0) {
+        va_wayland_error("could not add listener to wl_registry");
+        goto end;
+    }
     if (!wayland_roundtrip_queue(ctx->native_dpy, wl_drm_ctx->queue))
-        return false;
+        goto end;
 
     /* registry_handle_global should have been called by the
      * wl_display_roundtrip_queue above
      */
 
     /* Do not print an error, the compositor might just not support wl_drm */
     if (!wl_drm_ctx->drm)
-        return false;
+        goto end;
 
-    wl_drm_add_listener(wl_drm_ctx->drm, &drm_listener, pDisplayContext);
+    if (wl_drm_add_listener(wl_drm_ctx->drm, &drm_listener, pDisplayContext) != 0) {
+        va_wayland_error("could not add listener to wl_drm");
+        goto end;
+    }
     if (!wayland_roundtrip_queue(ctx->native_dpy, wl_drm_ctx->queue))
-        return false;
+        goto end;
     if (drm_state->fd < 0) {
         va_wayland_error("did not get DRM device");
-        return false;
+        goto end;
     }
 
     if (!wayland_roundtrip_queue(ctx->native_dpy, wl_drm_ctx->queue))
-        return false;
+        goto end;
 
     if (!wl_drm_ctx->is_authenticated) {
         va_wayland_error("Wayland compositor did not respond to DRM authentication");
-        return false;
+        goto end;
+    }
+
+    result = true;
+
+end:
+    if (wrapped_display) {
```

Comment:
It is possible wrapped_display is not initialized.   please set wrapped_display to NULL at the beginning.